consumer/file_writers/minio.py

```diff
@@ -0,0 +1,120 @@
+import os
+from io import BytesIO
+
+from minio import Minio
+from minio.error import S3Error
+
+from common.logger import get_logger
+from config.config import MinioProperties, get_consumer_config
+from consumer.file_writers.base import FileWriterBase, get_list_hash
+
+logger = get_logger("minio-file-writer")
+
+
+def create_minio_client(minio_config: MinioProperties | None = None) -> Minio:
+    """
+    Creates and returns a Minio client using the provided configuration or
+    default configuration if none is provided.
+
+    Returns:
+        Minio: An initialized Minio client.
+    """
+    if not minio_config:
+        config = get_consumer_config()
+        minio_config = config.minio
+    minio_config_dict = minio_config.model_dump(by_alias=True)
+    logger.debug(f"Minio config raw type: {type(minio_config_dict)}, data: {minio_config_dict}")
+    minio_client = Minio(**minio_config_dict)
+    return minio_client
+
+
+class MinioFileWriter(FileWriterBase):
+    """
+    A file writer that saves messages to MinIO storage.
+
+    Args:
+        root_path (str): MinIO bucket name.
+        minio_client (Minio): Minio client instance.
+        NOTE: password and login should be stored as environment vars (see Readme)
+    """
+
+    def __init__(self, root_path: str, minio_client: Minio) -> None:
+        self._root_path = root_path  # This is minio bucket_name
+        self._minio_client = minio_client
+
+    def create_bucket(self, bucket_name: str | None = None) -> None:
```

Comment:
create bucket moved inside the class, since it anyway needs minio client to execute, no need to pass it in the function arguments